support/cas-server-support-token-tickets/src/main/java/org/apereo/cas/token/authentication/principal/TokenWebApplicationServiceResponseBuilder.java

```diff
@@ -40,11 +39,14 @@ protected WebApplicationService buildInternal(final WebApplicationService servic
         val registeredService = this.servicesManager.findServiceBy(service);
         RegisteredServiceAccessStrategyUtils.ensureServiceAccessIsAllowed(service, registeredService);
         val tokenAsResponse = RegisteredServiceProperty.RegisteredServiceProperties.TOKEN_AS_SERVICE_TICKET.isAssignedTo(registeredService);
+        val ticketIdAvailable=isTicketIdAvailable(parameters);
 
-        if (!tokenAsResponse) {
-            LOGGER.debug("Registered service [{}] is not configured to issue JWTs for service tickets. "
-                    + "Make sure the service property [{}] is defined and is set to true", registeredService,
-                RegisteredServiceProperties.TOKEN_AS_SERVICE_TICKET.getPropertyName());
+        if (!tokenAsResponse||!ticketIdAvailable) {
+            if (ticketIdAvailable) {
+                LOGGER.debug("Registered service [{}] is not configured to issue JWTs for service tickets. "
+                                + "Make sure the service property [{}] is defined and set to true", registeredService,
+                        RegisteredServiceProperties.TOKEN_AS_SERVICE_TICKET.getPropertyName());
+            }
             return super.buildInternal(service, parameters);
```

Comment:
Fix the formatting issues here. Spaces between operators, and also fix the indentation please.

Reply:
Done.